### PR TITLE
Implement ExactSizeIterator for ToLowercase and ToUppercase

### PR DESCRIPTION
**Pull request by sugar700**

### CODE REVIEW

File: src/libcore/tests/char.rs [hunk 1]
```diff
@@ -76,6 +76,8 @@ fn test_to_digit() {
 #[test]
 fn test_to_lowercase() {
     fn lower(c: char) -> String {
+        let to_lowercase = c.to_uppercase();
```
Comment (ollie27) [inline src/libcore/tests/char.rs]:
```suggestion
        let to_lowercase = c.to_lowercase();
```

**Comment by Dylan-DPC** [comment]
ping from triage can anyone from @rust-lang/libs review this?

**Comment by SimonSapin** [comment]
Looks good, thanks!

@bors r+

**Comment by kennytm** [comment]
r? @SimonSapin needs reapproval after force-push.

BTW does this PR still "needs-fcp"?

**Comment by SimonSapin** [comment]
We usually do FCP when adding new stable APIs. But in this case I’m gonna unilaterally declare this obvious enough. CC @rust-lang/libs

@bors r+

**Pull request merged**